### PR TITLE
Backend HFI calculations hooked up on frontend

### PR DESCRIPTION
**Pull request by conbrad**
- Frontend hooked up to backed HFI calculations 
- `selectedPrepDay` remains a redux state since we don't need to make new backend calls for daily/prep week tabs
- Removed frontend CSV code in anticipation of PDF generation 

# Test Links:
[Percentile Calculator](https://wps-pr-1722.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1722.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1722.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1722.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1722.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=1108&f=c5&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN#state=2ec784ca-c46a-49d0-b2b3-1cf32a9015a2&session_state=7d9447c8-db66-4661-b4cb-03d2ac0d1d8f&code=32292df4-2bdf-4f90-a4a8-c8dbcda682a9.7d9447c8-db66-4661-b4cb-03d2ac0d1d8f.2b63f390-f3dc-43ae-89f2-016453863476)
[Fire Behaviour Advisory](https://wps-pr-1722.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1722.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: web/src/api/hfiCalculatorAPI.ts [hunk 1]
```diff
@@ -71,3 +77,43 @@ export async function getDailies(
     last_updated: DateTime.fromISO(daily.last_updated)
   }))
 }
+
+export async function getHFIResult(
+  request: HFIResultRequest
+): Promise<HFIResultResponse> {
+  const { data } = await axios.post<RawHFIResultResponse>(baseUrl, {
+    ...request,
+    selected_prep_date: request.selected_prep_date?.toISOString().split('T')[0] // Infinite sadness
```
Comment (Sybrand) [inline web/src/api/hfiCalculatorAPI.ts:86]: Only used once, by maybe wrapping in function and explaining in detail what's going on here? We have so many time-zone conversations, I can only assume that whomever has to maintain this code will be lost without context.

File: web/src/api/hfiCalculatorAPI.ts [hunk 2]
```diff
@@ -71,3 +77,43 @@ export async function getDailies(
...
+    data.planning_area_hfi_results.map(areaResult => ({
+      ...areaResult,
+      daily_results: areaResult.daily_results.map(dr => ({
+        ...dr,
+        dailies: dr.dailies.map(validatedDaily => ({
+          ...validatedDaily,
+          daily: {
+            ...validatedDaily.daily,
+            date: DateTime.fromISO(
+              validatedDaily.daily.date.split('T')[0] + 'T' + '00:00-08:00' // Infinite sadness
```
Comment (Sybrand) [inline web/src/api/hfiCalculatorAPI.ts:103]: Same as above - re function and comment (except this is re-used, so function makes more sense). Also the 8 hour offset we should be using a constant.

**Review by Sybrand — CHANGES_REQUESTED** [review]
The only good feedback I have is on the insidious date matter.
There's obviously a lot of scope for re-factoring - but this PR is already pretty big. Well done on getting everything working!

**Review by JulianForeman — APPROVED** [review]
Had to google what clonedeep did but lgtm

File: web/src/api/hfiCalculatorAPI.ts (hunk 2 above)
Comment (andrea-williams) [reply web/src/api/hfiCalculatorAPI.ts:103]: all that nested mapping is infinite sadness too :(

File: web/src/features/hfiCalculator/components/FireStartsDropdown.tsx [hunk 3]
```diff
@@ -10,14 +10,14 @@ import React from 'react'
 
 export interface FireStartsDropdownProps {
   fireStarts: FireStarts | undefined
-  areaName: string
+  areaId: number
   dayOffset: number
-  setFireStarts: (areaName: string, dayOffset: number, newFireStarts: FireStarts) => void
+  setFireStarts: (areaId: number, dayOffset: number, newFireStarts: FireStarts) => void
```
Comment (andrea-williams) [inline web/src/features/hfiCalculator/components/FireStartsDropdown.tsx:15]: oooh yes I much prefer using areaId instead of areaName!!

**Review by andrea-williams — APPROVED** [review]
Gonna say approved - I read through the code and didn't spot anything that's an easy fix in terms of refactoring. Haven't been able to view the site in dev because I still don't have home internet and something's real weird about not letting me access any of our apps from a tethered data network.

File: web/src/api/hfiCalcAPI.ts [hunk 4]
```diff
@@ -15,14 +15,14 @@ export interface WeatherStationProperties {
 export interface FireCentre {
   id: number
   name: string
-  planning_areas: Record<string, PlanningArea>
+  planning_areas: PlanningArea[]
```
[the remark this answers is not in the document]
Comment (conbrad) [reply web/src/api/hfiCalcAPI.ts:18]: The backend returns lists now, and it's much nicer to work with lists compared to records when we mostly iterate over them.

File: web/src/features/hfiCalculator/components/DailyViewTable.tsx [hunk 5]
```diff
@@ -289,192 +291,190 @@ export const DailyViewTable = (props: Props): JSX.Element => {
...
+                      areaName={area.name}
+                      fireStarts={dailyResult?.fire_starts}
+                    />
+                    <PrepLevelCell
+                      testid={`daily-prep-level-${area.id}`}
+                      toolTipText=" Incomplete data from WFWX for one or more stations. Please exclude station(s) displaying errors."
+                      prepLevel={dailyResult?.prep_level}
+                    />
+                  </TableRow>
+                  {sortBy(area.stations, station => station.code).map(station => {
```
Comment (andrea-williams) [inline web/src/features/hfiCalculator/components/DailyViewTable.tsx:356]: Shouldn't be sorting by station code, should be sorting by station's `order_of_appearance_in_planning_area_list`

**Review by andrea-williams — CHANGES_REQUESTED** [review]
Just need to change the way stations get sorted in planning areas

**Pull request merged**